transformer_engine/pytorch/constants.py

```diff
@@ -29,4 +30,4 @@
 
 GemmParallelModes = ("row", "column", None)
 
-dist_group_type = torch._C._distributed_c10d.ProcessGroup
+dist_group_type = torch.distributed.distributed_c10d.ProcessGroup
```

Comment:
We should prefer using PyTorch's public API: 
```suggestion
dist_group_type = torch.distributed.ProcessGroup
```
We should also import `torch.distributed` instead of `torch.distributed.distributed_c10d`.